lib/nmatrix/shortcuts.rb

```diff
@@ -236,6 +236,86 @@ def eye(shape, opts={})
       m
     end
     alias :identity :eye
+
+    #
+    # call-seq:
+    #     hilbert(shape) -> NMatrix
+    #     hilbert(shape, dtype: dtype) -> NMatrix
+    #     hilbert(shape, stype: stype, dtype: dtype) -> NMatrix
+    #
+    # Creates an hilbert matrix (square matrix).
+    #
+    # * *Arguments* :
+    #   - +size+ -> integer ( for square matrix) specifying the dimensions.
+    #   - +dtype+ -> (optional) Default is +:float64+
+    #   - +stype+ -> (optional) Default is +:dense+.
+    # * *Returns* :
+    #   - A hilbert matrix.
+    #
+    # Examples:
+    #
+    #    NMatrix.hilbert(3) # =>  1.0     0.5      0.3333333333333333
+    #            0.5                         0.3333333333333333    0.25
+    #            0.3333333333333333          0.25                  0.2
+    #
+    def hilbert(shape, opts={})
+      # Fill the diagonal with 1's.
+      m = NMatrix.ones(shape, {:dtype => :float64}.merge(opts))
+      (0...(shape)).each do |i|
+          (0...(i+1)).each do |j|
+            m[i,j]=m[i,j]/(j+i+1)
+            if i!=j
+              m[j,i]=m[i,j]
+            end
+          end
+       end
+      m
```

Comment:
See my comments for the `inv_hilbert` function. Same applies to this function too.